airflow/providers/amazon/aws/example_dags/example_glue.py

```diff
@@ -0,0 +1,139 @@
+from os import getenv
+
+from airflow import DAG
+from airflow.models.baseoperator import chain
+from airflow.operators.python import PythonOperator
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.amazon.aws.operators.glue import GlueJobOperator
+from airflow.providers.amazon.aws.operators.glue_crawler import GlueCrawlerOperator
+from airflow.providers.amazon.aws.sensors.glue import GlueJobSensor
+from airflow.providers.amazon.aws.sensors.glue_crawler import GlueCrawlerSensor
+
+GLUE_DATABASE_NAME = getenv('GLUE_DATABASE_NAME', 'glue_database_name')
+GLUE_EXAMPLE_S3_BUCKET = getenv('GLUE_EXAMPLE_S3_BUCKET', 'glue_example_s3_bucket')
+
+# Role needs putobject/getobject access to the above bucket as well as the glue
+# service role, see docs here: https://docs.aws.amazon.com/glue/latest/dg/create-an-iam-role.html
+GLUE_CRAWLER_ROLE = getenv('GLUE_CRAWLER_ROLE', 'glue_crawler_role')
+GLUE_CRAWLER_NAME = 'example_crawler'
+GLUE_CRAWLER_CONFIG = {
+    'Name': GLUE_CRAWLER_NAME,
+    'Role': GLUE_CRAWLER_ROLE,
+    'DatabaseName': GLUE_DATABASE_NAME,
+    'Targets': {
+        'S3Targets': [
+            {
+                'Path': f'{GLUE_EXAMPLE_S3_BUCKET}/input',
+            }
+        ]
+    },
+}
+
+# Example csv data used as input to the example AWS Glue Job.
+EXAMPLE_CSV = '''
+food,price
+apple,0.5
+milk,2.5
+bread,4.0
+'''
+
+# Example Spark script to operate on the above sample csv data.
+EXAMPLE_SCRIPT = f'''
+from pyspark.context import SparkContext
+from awsglue.context import GlueContext
+
+glueContext = GlueContext(SparkContext.getOrCreate())
+datasource = glueContext.create_dynamic_frame.from_catalog(
+             database='{GLUE_DATABASE_NAME}', table_name='input')
+print('There are %s items in the table' % datasource.count())
+
+datasource.toDF().write.format('csv').mode("append").save('s3://{GLUE_EXAMPLE_S3_BUCKET}/output')
+'''
+
+
+def _upload_from_tmp_file_to_glue_bucket(contents: str, s3_key: str):
+    '''Upload contents of string to S3 leveraging tempfiles to copy the data'''
+    with tempfile.NamedTemporaryFile(mode='a') as tmp:
+        tmp.write(contents.strip())
+        tmp.seek(0)
+        s3_hook = S3Hook()
+        s3_hook.load_file(
+            filename=tmp.name,
+            key=s3_key,
+            bucket_name=GLUE_EXAMPLE_S3_BUCKET,
+            replace=True,
+        )
+
+
+with DAG(
+    dag_id='example_glue',
+    schedule_interval=None,
+    start_date=datetime(2021, 1, 1),
+    tags=['example'],
+    catchup=False,
+) as glue_dag:
+
+    upload_csv = PythonOperator(
+        task_id='upload_csv',
+        python_callable=_upload_from_tmp_file_to_glue_bucket,
+        op_kwargs={'contents': EXAMPLE_CSV, 's3_key': 'input/input.csv'},
+    )
+    upload_etl_script = PythonOperator(
+        task_id='upload_etl_script',
+        python_callable=_upload_from_tmp_file_to_glue_bucket,
+        op_kwargs={'contents': EXAMPLE_SCRIPT, 's3_key': 'etl_script.py'},
+    )
```

Comment:
We can have less code lines with the python decorator

Reply:
We really should be using the TaskFlow API where possible. There was an effort a while back to transition the example DAGs to this approach. Related to #9415

Reply:
Agreed.

Reply:
Sure, folks, I will make the change :)